js/app/demo/rulesConfig.jsx

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+define(['react'], function(React) {
+    return React.createClass({
+        displayName: 'RulesConfig',
+        getInitialState: function() {
+            return this.props.config;
+        },
+        buildRules: function(state) {
```

Comment:
Sure. Incomplete refactoring:-) This was originally private function, and I didn't fully refactor it. I have to say, JSX isn't all that much fun to deal with when you have to do complex looping... Simple `map` is fine and comfortable, but if you need to do something more complex in a loop, it sort-of becomes a mess pretty quickly.